src/Http/Controllers/OAuthController.php

```diff
@@ -17,7 +18,14 @@ public function redirectToProvider($provider)
     public function handleProviderCallback($provider)
     {
         try {
-            $providerUser = Socialite::driver($provider)->user();
+            $socialiteProvider = Socialite::driver($provider);
+            $providerConfig = OAuth::providers()->first(fn ($prov) => $provider === $prov->name())->config();
+
+            if (Arr::get($providerConfig, 'stateless', false)) {
+                $socialiteProvider->stateless();
+            }
+
+            $providerUser = $socialiteProvider->user();
```

Comment:
Can we refactor this so you can do something like

```php
$provider = OAuth::provider($provider); // this happens in a few lines anyway

$providerUser = $provider->getSocialiteUser();
```

```php
class Provider
{
  function getSocialiteUser()
  {
    $driver = Socialite::driver($this->name);
    
    if ($this->config['stateless']) {
      $driver->stateless();
    }

    return $driver->user();
  }
}
```

(Written without testing so don't just copy/paste) 😀